### PR TITLE
fix: File Upload description bypassing message maximum character limit setting

### PR DESCRIPTION
**Pull request by MartinSchoeler**
## Proposed changes (including videos or screenshots)

Added the message character limit restriction to the file upload description field, before this, you could send a message bigger than the limit by sending a file with a large description

## Issue(s)

[SUP-620](https://rocketchat.atlassian.net/browse/SUP-620)

[SUP-620]: https://rocketchat.atlassian.net/browse/SUP-620?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by abhinavkrin — CHANGES_REQUESTED** [review]
Hello @MartinSchoeler,

Would it be a good idea to implement the check on the backend as well? Currently, users can send long messages through the API bypassing the character limit.

**Review by abhinavkrin — DISMISSED** [review]

**Review by tiagoevanp — DISMISSED** [review]

**Review by gabriellsh — CHANGES_REQUESTED** [review]
The title should describe the issue and not the solution. I'll change that for you.

**Review by gabriellsh — DISMISSED** [review]

File: apps/meteor/tests/e2e/file-upload.spec.ts [hunk 1]
```diff
@@ -76,6 +76,7 @@ test.describe.serial('file-upload', () => {
 		await expect(poHomeChannel.content.btnModalConfirm).not.toBeVisible();
 	});
 });
+
```
Comment (MarcosSpessatto) [inline apps/meteor/tests/e2e/file-upload.spec.ts:79]:
```suggestion
```

File: apps/meteor/app/api/server/v1/rooms.ts
Comment (MarcosSpessatto) [inline apps/meteor/app/api/server/v1/rooms.ts:1]: I think we need to add the same validation for the `rooms.upload` endpoint, even if it's deprecated...

**Review by MarcosSpessatto — CHANGES_REQUESTED** [review]
Can we add some UI

File: apps/meteor/tests/e2e/file-upload.spec.ts (hunk 1 above)
Comment (MartinSchoeler) [reply apps/meteor/tests/e2e/file-upload.spec.ts:79]: https://github.com/RocketChat/Rocket.Chat/pull/33218#discussion_r1811521745

**Review by MarcosSpessatto — APPROVED** [review]

**Review by dougfabris — APPROVED** [review]
Great work here!

**Pull request merged**